app/src/main/java/com/github/se/eventradar/ui/event/EventDetails.kt

```diff
@@ -71,22 +72,25 @@ fun EventDetails(viewModel: EventDetailsViewModel, navigationActions: Navigation
       },
       floatingActionButton = {
         // view ticket button
-        if (!isUserAttending) {
-          FloatingActionButton(
-              onClick = {
+        val icon = if (isUserAttending) R.drawable.cancel else R.drawable.ticket
+        FloatingActionButton(
+            onClick = {
+              if (isUserAttending) {
+                viewModel.showCancelRegistrationDialog.value = true
+              } else {
                 navigationActions.navController.navigate(
                     "${Route.EVENT_DETAILS_TICKETS}/${viewModel.eventId}")
-              },
-              modifier = Modifier.testTag("ticketButton"),
-              containerColor = MaterialTheme.colorScheme.primaryContainer,
-          ) {
-            Icon(
-                painter = painterResource(id = R.drawable.ticket),
-                contentDescription = "view tickets button",
-                modifier = Modifier.size(32.dp),
-                tint = MaterialTheme.colorScheme.onPrimaryContainer,
-            )
-          }
+              }
+            },
+            modifier = Modifier.testTag("ticketButton"),
+            containerColor = MaterialTheme.colorScheme.primaryContainer,
```

Comment:
nit : should we rename the test tag `registrationButton` instead of `ticketButton` since it now take care of joining and leaving event ?